ring-jetty-adapter/src/ring/adapter/jetty.clj

```diff
@@ -150,10 +172,15 @@
 (defn- ^Server create-server [options]
   (let [pool   (or (:thread-pool options) (create-threadpool options))
         server (Server. pool)]
-    (when (:http? options true)
+    (when (:http? options (or ;; default is enabled when no socket specified
+                              (not (:unix-socket options))
+                              ;; enabled with socket when host/port set
+                              (some options [:host :port])))
```

Comment:
This feels too complex, and isn't what the documentation says. I think let's keep it simple and just leave it defaulting to true. Let's add a note to the `:unix-socket` option instead to tell people they may want to set `:http?` to false.